mlr-org/gallery/technical/2023-10-30-tidymodels/index.qmd

```diff
@@ -0,0 +1,847 @@
+  digits = 0,
+  col.names = c("Resampling", "LQ", "Median", "UQ"))
+```
+
+:::
+
+The utilization of the `doParallel` package demonstrates a notable improvement in handling smaller resampling tasks.
+In these scenarios, the resampling process consistently outperforms the `mlr3` framework in terms of speed.
+However, it's important to note that even with this enhanced performance, the `doParallel` package does not always surpass the efficiency of the sequential version, especially when working with the `rpart` model.
+This specific observation is illustrated in the left section of @fig-resample-parallel.
+
+:::{layout-ncol="2"}
+
+```{r}
+#| echo: false
+#| label: tbl-resample-parallel-tidymodels-parallel-rpart
+#| tbl-cap: "Average runtime in milliseconds of `tidymodels` with `doParallel`  and `rpart` depending on the resampling strategy."
+table = setDT(summary(bm_rpart$bm_2_3))
+table[, framework := str_split_fixed(expr, "_", 2)[, 1]]
+table[, resampling := str_split_fixed(expr, "_", 2)[, 2]]
+
+knitr::kable(
+  x = table[, list(resampling, lq, median, uq)],
+  digits = 0,
+  col.names = c("Resampling", "LQ", "Median", "UQ"))
+```
+
+```{r}
+#| echo: false
+#| label: tbl-resample-parallel-tidymodels-parallel-ranger
+#| tbl-cap: "Average runtime in milliseconds of `tidymodels` with `doParallel`  and `ranger` depending on the resampling strategy."
+table = setDT(summary(bm_ranger$bm_2_3))
+table[, framework := str_split_fixed(expr, "_", 2)[, 1]]
+table[, resampling := str_split_fixed(expr, "_", 2)[, 2]]
+
+knitr::kable(
+  x = table[, list(resampling, lq, median, uq)],
+  digits = 0,
+  col.names = c("Resampling", "LQ", "Median", "UQ"))
+```
+
+:::
+
+
+```{r}
+#| echo: false
+#| label: fig-resample-parallel
+#| fig-cap: "Average runtime, measured in milliseconds, for cross-validations using `rpart` (displayed on the left) and `ranger` (on the right). The comparison encompasses variations across different frameworks, the number of folds in the cross-validation, and the implementation of parallelization."
+res_2 = setNames(map(list(bm_rpart$bm_2_1, bm_rpart$bm_2_2, bm_rpart$bm_2_3), as.data.table), c("mlr3_future", "tidymodels_future", "tidymodels_parallel"))
+res_2 = rbindlist(res_2, idcol = "framework")
+res_2[c("mlr3_cv3", "tidymodels_cv3"), folds := 3, on = "expr"]
+res_2[c("mlr3_cv6", "tidymodels_cv6"), folds := 6, on = "expr"]
+res_2[c("mlr3_cv9", "tidymodels_cv9"), folds := 9, on = "expr"]
+res_2[c("mlr3_rcv100", "tidymodels_rcv100"), folds := 100, on = "expr"]
+res_2[, folds := factor(folds)]
+res_2[, framework := factor(framework)]
+res_2 = res_2[, list(time = median(time)), by = c("expr", "framework", "folds")]
+res_2[, time := time / 1e+6]
+
+res_rpart_1_2 = rbindlist(list(
+  sequential = res_1_rpart,
+  parallel = res_2),
+  use.names = TRUE, idcol = "mode")
+
+p1 = ggplot(res_rpart_1_2[c("mlr3_cv3", "tidymodels_cv3", "mlr3_cv6", "tidymodels_cv6", "mlr3_cv9", "tidymodels_cv9"), , on = "expr"],
```

Comment:
in these plots I would also add plot subtitles (rpart) and (ranger)

Reply:
Unfortunately this is a cowplot to have shared y-axis but i will add a title to each plot